sharness/t0050-service-state-upgrade-from-current.sh

```diff
@@ -0,0 +1,24 @@
+#! /bin/sh
+
+test_description="Test service state 'upgrade' from current version"
+
+. lib/test-lib.sh
+
+test_ipfs_init
+cleanup test_clean_ipfs
+test_cluster_init
+cleanup test_clean_cluster
+
+test_expect_success IPFS,CLUSTER "cluster-service state preserved by migration" '
+    cid=`docker exec ipfs sh -c "echo testing | ipfs add -q"` &&
+    ipfs-cluster-ctl pin add "$cid" && sleep 2 &&
+    kill -1 $CLUSTER_D_PID && sleep 30 &&
+    ipfs-cluster-service --config "test-config" state upgrade &&
+    ipfs-cluster-service --config "test-config" >"$IPFS_OUTPUT" 2>&1 &
+    sleep 50 &&
```

Comment:
I have had a lot of trouble with cluster starting up slowly the second time around and needing this long sleep.  It may just be my local setup so I will try it out.